### PR TITLE
feat(CB2-7471): trigger trailer into service generation

### PR DESCRIPTION
**Pull request by leifkemp-bjss**
## CB2-7471: Trigger Trailer into Service Generation

_PR to trigger generation of trailer into service letters in VTM_
[CB2-7471](https://dvsa.atlassian.net/browse/CB2-7471)

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

File: src/app/features/tech-record/components/tech-record-generate-letter/tech-record-generate-letter.component.ts [hunk 1]
```diff
@@ -44,30 +57,41 @@ export class GenerateLetterComponent implements OnInit {
...
-  ngOnInit(): void {
-    if (!this.currentTechRecord || this.currentTechRecord.vehicleType !== 'trl') {
-      this.navigateBack();
-    }
-
-    this.actions$.pipe(ofType(generateLetterSuccess), take(1)).subscribe(() => this.navigateBack());
+  get mostRecentLetter(): LettersOfAuth | undefined {
+    return (
+      this.currentTechRecord &&
+      cloneDeep(this.currentTechRecord.lettersOfAuth)
```
Comment (naathanbrown) [inline src/app/features/tech-record/components/tech-record-generate-letter/tech-record-generate-letter.component.ts:68]: lettersOfAuth is an object, not an array, you will only ever have the most recent

File: src/app/forms/custom-sections/letters/letters.component.ts [hunk 2]
```diff
@@ -46,7 +46,18 @@ export class LettersComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   get mostRecentLetter(): LettersOfAuth | undefined {
-    return this.vehicleTechRecord.lettersOfAuth && this.vehicleTechRecord.lettersOfAuth[this.vehicleTechRecord.lettersOfAuth.length - 1];
+    return cloneDeep(this.vehicleTechRecord.lettersOfAuth)
+      ?.sort((a, b) =>
+        a.letterDateRequested && b.letterDateRequested ? new Date(a.letterDateRequested).getTime() - new Date(b.letterDateRequested).getTime() : 0
```
Comment (naathanbrown) [inline src/app/forms/custom-sections/letters/letters.component.ts:51]: same here

File: src/app/models/vehicle-tech-record.model.ts [hunk 3]
```diff
@@ -138,12 +137,49 @@ export enum VehicleSubclass {
   W = 'w'
 }
```
Comment (naathanbrown) [inline src/app/models/vehicle-tech-record.model.ts:139]: Need to go in here and set tech record to use a single object opposed to array as well

File: src/app/features/tech-record/components/tech-record-generate-letter/tech-record-generate-letter.component.ts (hunk 1 above)
Comment (leifkemp-bjss) [reply src/app/features/tech-record/components/tech-record-generate-letter/tech-record-generate-letter.component.ts:68]: It's currently stored as an array in VTM but I can make that change

**Pull request merged**